server/src/main/java/org/elasticsearch/search/sort/FieldSortBuilder.java

```diff
@@ -359,6 +372,81 @@ public SortFieldAndFormat build(QueryShardContext context) throws IOException {
         return new SortFieldAndFormat(field, fieldType.docValueFormat(null, null));
     }
 
+    /**
+     * Return true if the primary sort in the provided <code>source</code>
+     * is an instance of {@link FieldSortBuilder}.
+     */
+    public static boolean hasPrimaryFieldSort(SearchSourceBuilder source) {
+        return getPrimaryFieldSortOrNull(source) == null;
+    }
+
+    /**
+     * Return the {@link FieldSortBuilder} if the primary sort in the provided <code>source</code>
+     * is an instance of this class, null otherwise.
+     */
+    public static FieldSortBuilder getPrimaryFieldSortOrNull(SearchSourceBuilder source) {
+        if (source == null || source.sorts() == null || source.sorts().isEmpty()) {
+            return null;
+        }
+        return source.sorts().get(0) instanceof FieldSortBuilder ? (FieldSortBuilder) source.sorts().get(0) : null;
+    }
+
+    /**
+     * Return the maximum {@link Comparable} indexed value from the provided {@link FieldSortBuilder} or <code>null</code> if the
+     * value is unknown. The value can be extracted on non-nested indexed mapped fields of type keyword, numeric or date, other field
+     * and configuration return <code>null</code>
+     * NOTE: The maximum value is the minimum if the order of the provided {@link FieldSortBuilder} is {@link SortOrder#DESC}.
+     */
+    public static Comparable getMaxSortValueOrNull(QueryShardContext context, FieldSortBuilder sortBuilder) throws IOException {
```

Comment:
maybe s/Max/Best/ since Max might be confusing for ascending sorts

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/49092/commits/9f959dc1db43577a02c1f5b9864c211e235db5e2 to return the min and the max values instead.